go/vt/mysqlctl/builtinbackupengine.go

```diff
@@ -702,15 +705,36 @@ func (be *BuiltinBackupEngine) backupFile(ctx context.Context, params BackupPara
 	// Create the gzip compression pipe, if necessary.
 	var compressor io.WriteCloser
 	if backupStorageCompress {
+		// Track the duration of the underlying writer, so that we can
+		// calculate the duration of compression by itself.
+		timedWriter := ioutil.NewTimedWriter(writer)
+
 		if ExternalCompressorCmd != "" {
-			compressor, err = newExternalCompressor(ctx, ExternalCompressorCmd, writer, params.Logger)
+			compressor, err = newExternalCompressor(ctx, ExternalCompressorCmd, timedWriter, params.Logger)
 		} else {
-			compressor, err = newBuiltinCompressor(CompressionEngineName, writer, params.Logger)
+			compressor, err = newBuiltinCompressor(CompressionEngineName, timedWriter, params.Logger)
 		}
 		if err != nil {
 			return vterrors.Wrap(err, "can't create compressor")
 		}
-		writer = compressor
+
+		// The compression duration includes the time it takes to write
```

Comment:
I didn't have to do this. We could just report compression time without trying to factor out that because of the way we're chaining `io.Writer` together that that time includes disk writing (or S3 uploading) time. In other words we could leave it as an exercise for the Grafana use to subtract one metric from another to make meaning out of them.

I prefer this, but don't feel strongly about it.